### DISCUSSION TITLE
How do I pull a new version?

### DISCUSSION THREAD

**Discussion by BEisem** [Taxel/PlexTraktSync#236] [Q&A] [answered]
Sorry because I know this is a github question and not specific to this project. But I don't know the best way to update to a new version.

I previously used `git clone -b 0.6.44 --depth=1 https://github.com/Taxel/PlexTraktSync` to clone the (at the time) most recent version for use on my local machine. Now I am trying to update to the newer 0.7.4. What command would I use to do that?

**Comment by glensc** [comment, marked as answer]
replace `0.6.44` with `0.7.4`?

why can't you read out the instructions from here:
- `https://github.com/Taxel/PlexTraktSync/pull/195/files`

----

1. visit releases
2. pick release
3. download it!

please, suggest, in that very same pr how would you understand how to do that? because it's written there and yet you don't see that.

**Reply by BEisem** [reply]
I understand how to download to a new location. But if I'm trying to update my existing installation to the new one, it gives me errors. For example if I do `git clone -b 0.7.4 --depth=1 https://github.com/Taxel/PlexTraktSync` into my existing PlexTraktSync folder, it returns the following:

```bink@Homey-McPlexFace scripts % git clone -b 0.7.4 --depth=1 https://github.com/Taxel/PlexTraktSync
fatal: destination path 'PlexTraktSync' already exists and is not an empty directory.
```

According to `git --help`, the `clone` command is used to `Clone a repository into a new directory.` I was asking how to update my existing directory to a new version.

Sure, I could download the zip file and copy it, but then I also have to copy the existing `.env`, `.pytrakt.json` and `trakt_cache.sqlite.` I just thought there might be an easier way.

**Reply by glensc** [reply]
yes. you can do all that.

answer to the question, the best way: use the zip download! 

if using git, you need `git fetch --tags` and `git checkout` commands.

**Reply by flammable** [reply]
@BEisem It appears you can perform some automation around [this Stack Overflow answer](https://stackoverflow.com/a/22857288). I think this will be my approach.

Thank you both, I learned a lot here!

**Comment by flammable** [comment]
Hang on, so I shouldn't be doing a `git pull` on the master branch? I should be using the tags?

**Reply by BEisem** [reply]
> Hang on, so I shouldn't be doing a `git pull` on the master branch? I should be using the tags?

My understanding is that the original developer is non-responsive, so @glensc can't merge changes to the master. So the "master" branch is severely behind at this point. I'm currently using 0.7.9 (available via tags) and there are huge changes compared to the master branch.

**Reply by glensc** [reply]
you shouldn't use `master` in any case, in any project, as it's typically unreleased code.

I've added some documentation about this:
- https://github.com/Taxel/PlexTraktSync/pull/195

but it waits for approval from either taxel or twolaw to merge.

**Reply by glensc** [reply]
most of the refactoring happened on 0.6.x branch, making code more modular:
- https://github.com/Taxel/PlexTraktSync/pull/209

again waits for approval from either taxel or twolaw to merge.

I won't be making any changes to 0.6.x branch unless it's required for the application to function properly. so I consider the current stable to be 0.6.x. it does not mean that 0.7.x is unstable. 0.5.x could be considered oldstable.

**Reply by flammable** [reply]
Thanks! I've been using this project for about as long as you and @twolaw have (basically when @Taxel created the project), so `master` has always been the branch I've used. I'll switch to the version tags sourced from the `main` branch.

**Comment by glensc** [comment]
Found a [gh](https://github.com/cli/cli) tool to be useful here

```
brew install gh
gh repo clone Taxel/PlexTraktSync
cd PlexTraktSync
gh pr checkout 179
```

looks like even friendly to beginners
